lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/FastqsToBam.pm

```diff
@@ -57,6 +63,40 @@ sub execute {
     return 1;
 }
 
+sub _unarchive_fastqs_if_necessary {
+    my $self = shift;
+    $self->debug_message('Unarchive fastqs if necessary...');
+
+    my @new_fastq_paths;
+    for my $fastq_path ( $self->fastq_paths ) {
+        if ( $fastq_path !~ /\.gz$/ ) {
```

Comment:
@acoffman True, it is not the most reliable. If the file is gzipped and the extension is not .gz then gunzip will not work unless it is written to STDOUT. This just the simplest, and what Archive::Extract (I was originally going to use it) does.

@nnutter I am for that, but File::Type is not deployed for 5.10 on our system. We would need to hold this PR to do the testing and PR required to add this package. To note, the XTen imports are waiting on this PR.